### DISCUSSION TITLE
Favicons blocked with `no-large-media: * true` despite `no-large-media: behind-the-scene false`?

### DISCUSSION THREAD

**Discussion by tomasz1986** [uBlockOrigin/uBlock-issues#3348] [Q&A]
This happens with "Block media elements" set to "larger than 0 KB".

```
no-large-media: * true
no-large-media: behind-the-scene false
behind-the-scene * * noop
behind-the-scene * 1p-script noop
behind-the-scene * 3p noop
behind-the-scene * 3p-frame noop
behind-the-scene * 3p-script noop
behind-the-scene * image noop
behind-the-scene * inline-script noop
```

The goal is to block all images displayed on the actual site, yet I'd like to still allow uBlock to download and show favicons in the browser tab bar.

I've tested the same configuration with multiple websites in Chromium, Brave, and Firefox. There is no difference. With the media elements blocking set to "0 KB", favicons for all sites are blocked as well. The `no-large-media: behind-the-scene false` rule doesn't seem to have any effect on them.

Is this the expected behaviour?

**Comment by uBlock-user** [comment]
Favicons are images, so naturally they will also get blocked. As to whether they're pulled from `behind-the-scene` type network connection can only be found via the logger, which is something you will have to do it yourself.

> The goal is to block all images displayed on the actual site, yet I'd like to still allow uBlock to download and show favicons in the browser tab bar.

```
||*$image,domain=example.com
@@/favicon$image,domain=example.com
```

Use those filters.

**Reply by tomasz1986** [reply]
> > The goal is to block all images displayed on the actual site, yet I'd like to still allow uBlock to download and show favicons in the browser tab bar.
> 
> ```
> ||*$image,domain=example.com
> @@/favicon$image,domain=example.com
> ```
> 
> Use those filters.

Yeah, I know I can block all images like that but the problem is that in contrary to using the built-in image blocking functionality, this method doesn't allow to load individual images on demand on click 🙁. It also removes images with their `alt` text completely, so I cannot even know that they were there.

**Reply by uBlock-user** [reply]
> his method doesn't allow to load individual images on demand on click

Ofcourse that wouldn't be possible, because in this method, there's no connection made to the location of the image...so download is not possible.

**Reply by garry-ut99** [reply]
>  `@@/favicon$image,domain=example.com`

Also not all favicon images have name "favicon" or are in folder "favicon", hence such exception filter won't work in 100% of cases, for example on popular polish site `www.wp.pl`, favicon name is random-generated:
`https://www.wp.pl/eDE2LnBuSiUkUDpdbQ5HMWcIbkEkWldrdANiWXVbVWk1XSACbEVKIiREJQ0tG0h1c0p9WGwFCyNnTw`

**Reply by uBlock-user** [reply]
> not all favicon images have name "favicon" or are in folder "favicon",

my filters covers the majority of the cases, exceptions to the rule are to be expected but it depends on his browsing history.

**Reply by tomasz1986** [reply]
> Also not all favicon images have name "favicon" or are in folder "favicon", hence such exception filter won't work in 100% of cases, for example on popular polish site `www.wp.pl`, favicon name is random-generated:
> `https://www.wp.pl/eDE2LnBuSiUkUDpdbQ5HMWcIbkEkWldrdANiWXVbVWk1XSACbEVKIiREJQ0tG0h1c0p9WGwFCyNnTw`

Usually the old school `favicon.ico` is still present in the root, e.g. https://wp.pl/favicon.ico. I don't mind if some favicons are missing due to using funky filenames, but the problem here is that with no favicons displayed at all, it is quite impossible to use the browser normally, especially with many tabs open, where page titles are no longer displayed.

**Reply by garry-ut99** [reply]
`wp.pl/favicon.ico` isn't used as a favicon source on `www.wp.pl` , the random one is the one which is used, hence favicon is still blocked on `www.wp.pl` with his filters added.

**Reply by uBlock-user** [reply]
> hence favicon is still blocked on www.wp.pl with his filters added.

So create an exception? His browsing history do not include `wp.pl` unlike yours.

**Reply by garry-ut99** [reply]
> my filters covers the majority of the cases,

Not sure whether that majority is enought: it's 80% (checked top 100 websites), it requires 20 exceptions per every 100 sites visited, (means 200 per 1000), example sites it doesn't work on: facebook, instagram, whatsapp, discord...

Exceptions for popular favorite sites an user visits every day can be added in a short ime, but adding exceptions for random sites which an user visits daily is a constant work and might be too cumbersome (another idea is to create a public filter list "Favicons whitelist".)

> So create an exception?

Then you might end adding exceptions forever, because it changes every moment, now it's: 
`https://www.wp.pl/MTZ4MTYuYjU7Vjtndg5vIXgOb3s_Wn97awVjY25bfHs8VTs9OhojeWsCNWVvWz06PRYw`

**Reply by uBlock-user** [reply]
What he wants is a click-to-load feature for images with favicons as exceptions.

**Reply by tomasz1986** [reply]
> What he wants is a click-to-load feature for images with favicons as exceptions.

That's true, I can't really block all images with no option to quickly load them on an per-image basis. One workaround is to set the block image size to something like "larger than 5KB", which seems to be enough to allow things like favicons to get through. The problem is that then possibly much larger images with no `Content-Length` defined will also be downloaded (as described in https://github.com/gorhill/uBlock/wiki/Per-site-switches#caveat).

**Reply by uBlock-user** [reply]
You need to find another extension that hides all images by default and then lazyloads them as you scroll or do any action.

**Reply by tomasz1986** [reply]
> You need to find another extension that hides all images by default and then lazyloads them as you scroll or do any action.

Unfortunately, I don't think any such extension exists. At least not among open source projects. I've only managed to find some very outdated ones, or extensions that simply utilise the browser built-in functionality to block or unblock all images on a per-site basis. uBlock seems the only extension that offers the on demand image load functionality.

**Reply by garry-ut99** [reply]
> uBlock-user : What he wants is a click-to-load feature for images with favicons as exceptions.

I do perfectly know what he wants, I only replied to your idea, I wasn't suggesting it as a solution for his issue, actually it was you who was suggesting it to him, not me.

> tomasz1986: That's true, I can't really block all images with no option to quickly load them on an per-image basis.

True what: that you agree on his whole reply to me, or that you agree only with a part that his method is insufficient as a solution to your issue. If the former: then read above - nowhere I said you should follow his method. If the latter, then fine.

<hr>

> tomasz1986: Unfortunately, I don't think any such extension exists.


I've found something, three extensions, you can test them, though I don't guarante they will satisfy you 100%,  (unfortunately it has turned out that none of them meet OP requirements):

<details>
<summary>SPOILER</summary>

- you can test them, though I don't guarante it will satisfy you 100%, 
- extensions are available on both Firefox and Chromium, 
- except hiding all images, they have also feature to unhide (and hide) single images on demand
- 1 and 2 don't hide favicons in browser tabs, the 3rd does however

**1) Image hider** :
- https://chromewebstore.google.com/detail/image-hider/fdjghmbmljbhojdgegmhhbkhgoacgdcb

Some info:
- very easy to unhide/hide a single image on demand: just hover a mouse cursor over image
however I see also its drawback: it might lead to accidentaly unhiding unwanted images when moving mouse cursor over the page
- it doesn't block images, only hides them, hence no bandwidth saving
- sometimes images might appear (blink) for a while (fraction of second) before they are hidden

**2) Blur image and video** :
- https://chromewebstore.google.com/detail/blur-the-image-and-video/aikjogmpaoaookmacnkbenekcnkjlkmi
- https://addons.mozilla.org/pl/firefox/addon/blur-image-and-video

Some info:
- Alt+K - to blur/unblur howered image on demand, require two hands, unlike uBlock point and click,
but if you have mouse with many buttons you can assign the hot-key to a mouse button
-  I'am not sure whether the blurred images will be distracting for you or not, 
you can incerase blur from 50% to 100% to reduce blurred colors.
- it doesn't block images, only blurs them, hence no bandwidth saving

**3) Tunable image block** :
- https://chromewebstore.google.com/detail/tunable-image-block/agpjllgcceliiblmebkbiccaaldfedbc
- https://addons.mozilla.org/pl/firefox/addon/tunable-image-block

Some info:
- Use Ctrl + mouse move over image for load the image, it might not work perfectly.
- unfortunatelly it has turned out that this extension  hide favicons

</details>

**Reply by tomasz1986** [reply]
> > tomasz1986: That's true, I can't really block all images with no option to quickly load them on an per-image basis.
> 
> True what: that you agree on his whole reply to me, or that you agree only with a part that his method is insufficient as a solution to your issue. If the former: then read above - nowhere I said you should follow his method. If the latter, then fine.

"True" was about the fact that I would like "a click-to-load feature for images with favicons as exceptions" 😉. The whole method of using static filters to block images except for favicons isn't suitable for me, as that one doesn't allow to load each of them on demand.

> > tomasz1986: Unfortunately, I don't think any such extension exists.
> 
> I've found something, two extensions:

Thank you for the recommendations! The first extension is not for me, as my main reason to block images is to save bandwidth, but I've tested the second one. It does seem to work as advertised, however it also does block favicons here. It displays them if already cached, but if a completely new website is opened, the favicon is missing for it. I'm doing my testing in Chromium.

**Reply by garry-ut99** [reply]
> as my main reason to block images is to save bandwidth, 

Ok, I see, now I know, as you didn't mention about it before.

> but I've tested the second one., however it also does block favicons here. 

Yes, I confirm, unfortunately it does block them. 

<br>
It seems there are dozens of extensions to block images, but unfortunately it's difficult to find one which contains all the requested features, I don't know if such a one even exists, as I didn't test them all.

**Reply by tomasz1986** [reply]
It's all right. I don't need to block images all the time, but there are situations, where I have to use as little data as possible, so I will probably use uBlock to block them only then and just suffer the missing favicons 🤪.

**Comment by stephenhawk8054** [comment]
> With the media elements blocking set to "0 KB", favicons for all sites are blocked as well.

Yes, favicons are images that have size larger than 0KB, so of course they would be blocked. Using 0KB means blocking all images, including favicons.

> `no-large-media: behind-the-scene false`

`behind-the-scene` means [behind the scene network requests](https://github.com/gorhill/uBlock/wiki/Behind-the-scene-network-requests), it's for connections that cannot be associate with any tabs. If you mean the favicons of the website you are opening, those are not `behind-the-scene` connections.

**Reply by tomasz1986** [reply]
> > `no-large-media: behind-the-scene false`
> 
> `behind-the-scene` means [behind the scene network requests](https://github.com/gorhill/uBlock/wiki/Behind-the-scene-network-requests), it's for connections that cannot be associate with any tabs. If you mean the favicons of the website you are opening, those are not `behind-the-scene` connections.

Understood. I guess the old comment from https://github.com/gorhill/uBlock/issues/625#issuecomment-134155363 is no longer true then, as the current behaviour seems to be the same in Firefox as in the Chromium browsers.

**Reply by garry-ut99** [reply]
> tomasz1986: comment from https://github.com/gorhill/uBlock/issues/625#issuecomment-134155363

Also here some related information: https://github.com/gorhill/uBlock/issues/1732


> tomasz1986: The goal is to block all images displayed on the actual site, yet I'd like to still allow uBlock to download and show favicons in the browser tab bar.

Similiar request already declined in the past, your request can be considered a duplicate of:
- https://github.com/gorhill/uBlock/issues/3510
